File: core/src/com/unciv/game/pickerscreens/PickerScreen.java
package com.unciv.game.pickerscreens;

import com.badlogic.gdx.scenes.scene2d.InputEvent;
import com.badlogic.gdx.scenes.scene2d.ui.Label;
import com.badlogic.gdx.scenes.scene2d.ui.ScrollPane;
import com.badlogic.gdx.scenes.scene2d.ui.SplitPane;
import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.scenes.scene2d.ui.TextButton;
import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
import com.badlogic.gdx.utils.Align;
import com.unciv.game.CameraStageBaseScreen;
import com.unciv.game.UnCivGame;

public class PickerScreen extends CameraStageBaseScreen {

    TextButton closeButton;
    Label descriptionLabel;
    TextButton rightSideButton;
    float screenSplit = 0.85f;
    Table topTable;
    SplitPane splitPane;


    public PickerScreen(final UnCivGame game) {
        super(game);

        Table buttonTable = new Table();

        closeButton = new TextButton("Close", skin);
        closeButton.addListener(new ClickListener() {
            @Override
            public void clicked(InputEvent event, float x, float y) {
                game.setWorldScreen();
                dispose();
            }
        });
        buttonTable.add(closeButton).width(stage.getWidth() / 4);

        descriptionLabel = new Label("", skin);
        descriptionLabel.setWrap(true);
        descriptionLabel.setFontScale(game.settings.labelScale);
        descriptionLabel.setWidth(stage.getWidth() / 2);
        buttonTable.add(descriptionLabel).pad(5).width(stage.getWidth() / 2);

        rightSideButton = new TextButton("", skin);
        buttonTable.add(rightSideButton).width(stage.getWidth() / 4);
        buttonTable.setHeight(stage.getHeight()*(1-screenSplit));
        buttonTable.align(Align.center);

        topTable = new Table();
        ScrollPane scrollPane = new ScrollPane(topTable);

        scrollPane.setSize(stage.getWidth(), stage.getHeight() * screenSplit);

        splitPane = new SplitPane(scrollPane, buttonTable, true, skin);
        splitPane.setSplitAmount(screenSplit);
        splitPane.setFillParent(true);
        stage.addActor(splitPane);
    }
}

File: core/src/com/unciv/ui/pickerscreens/PickerScreen.kt
package com.unciv.ui.pickerscreens

import com.badlogic.gdx.scenes.scene2d.Actor
import com.badlogic.gdx.scenes.scene2d.ui.*
import com.unciv.Constants
import com.unciv.UncivGame
import com.unciv.ui.images.IconTextButton
import com.unciv.ui.utils.*
import com.unciv.ui.utils.AutoScrollPane as ScrollPane

open class PickerScreen(disableScroll: Boolean = false) : BaseScreen() {

    val pickerPane = PickerPane(disableScroll = disableScroll)

    /** @see PickerPane.closeButton */
    val closeButton by pickerPane::closeButton
    /** @see PickerPane.descriptionLabel */
    val descriptionLabel by pickerPane::descriptionLabel
    /** @see PickerPane.rightSideGroup */
    val rightSideGroup by pickerPane::rightSideGroup
    /** @see PickerPane.rightSideButton */
    val rightSideButton by pickerPane::rightSideButton

    /** @see PickerPane.topTable */
    val topTable by pickerPane::topTable
    /** @see PickerPane.scrollPane */
    val scrollPane by pickerPane::scrollPane
    /** @see PickerPane.splitPane */
    val splitPane by pickerPane::splitPane

    init {
        pickerPane.setFillParent(true)
        stage.addActor(pickerPane)
    }

    /**
     * Initializes the [Close button][closeButton]'s action (and the Back/ESC handler)
     * to return to the [previousScreen] if specified, or else to the world screen.
     */
    fun setDefaultCloseAction(previousScreen: BaseScreen?=null) {
        val closeAction = {
            if (previousScreen != null) game.setScreen(previousScreen)
            else game.setWorldScreen()
            dispose()
        }
        pickerPane.closeButton.onClick(closeAction)
        onBackButtonClicked(closeAction)
    }

    /** Enables the [rightSideButton]. See [pick] for a way to set the text. */
    fun setRightSideButtonEnabled(enabled: Boolean) {
        pickerPane.setRightSideButtonEnabled(enabled)
    }

    /** Sets the text of the [rightSideButton] and enables it if it's the player's turn */
    fun pick(rightButtonText: String) {
        pickerPane.pick(rightButtonText)
    }
}
